ZMQ.py

```diff
@@ -25,14 +23,14 @@ def listen(self):
             value = int(zmq_response[3])
             # catch tx for actual address
             if event == "tx":
-                if address == self.last_used_address:
-                    self.value = value
+                if address == self.state.last_used_address[0:81]:
                     self.state.tx_hash = zmq_response[1]
-                    if self.value != 0:
-                        self.state.set_state(1)
-                        print('Value TX incoming: ' + str(self.value) + ' iota')
+                    if value != 0:
+                        self.state.user_credit = value
```

Comment:
Better move this to `check_confirmation()` after `if conf_hash == self.state.tx_hash:`